hack/update-helm-index.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash -e
+
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o pipefail
+
+this=`basename $0`
+
+usage () {
+cat << EOF
+Usage: $this [-h]
+
+Options:
+  --helm-repo-path      specify the path to the Helm repo (defaults to HELM_REPO_PATH)
+  --help/-h             show this help and exit
+
+Example:
+  - from a local path:
+	$this /path/to/nvidia-device-plugin-0.15.0.tgz
+  - from a URL:	
+    $this https://github.com/NVIDIA/k8s-device-plugin/archive/refs/tags/nvidia-device-plugin-0.15.0.tgz
+
+EOF
+}
+
+#
+# Parse command line
+#
+while [[ $# -gt 0 ]]; do
+	key="$1"
+	case $key in
+		--helm-repo-path)
+			HELM_REPO_PATH="$2"
+			shift 2
+			;;
+		--help/h) usage
+			exit 0
+			;;
+		*) usage
+			exit 1
+			;;
+	esac
+done
+
+# now we take the input from the user and check if is a path or url http/https
+asset_path="$@"
+
+if [ -z "$asset_path" ]; then
+	echo "No assets provided"
+	exit 1
+fi
+
+if [[ $asset_path =~ ^https?:// ]]; then
+	asset_url=$asset_path
+else
+	asset_local=$asset_path
+fi
+
+GH_REPO_PULL_URL="https://github.com/NVIDIA/k8s-device-plugin.git"
+git clone --local $$(pwd) $(HELM_REPO_PATH)
+git -C $(HELM_REPO_PATH) remote add github $(GH_REPO_PULL_URL)
+git -C $(HELM_REPO_PATH) fetch github gh-pages
+git -C $(HELM_REPO_PATH) checkout gh-pages
+
+pushd $HELM_REPO_PATH/stable > /dev/null
+# Download charts from release assets
+
+if [ -n "$asset_local" ]; then
+	echo "Copying $asset_local..."
+	cp -f $asset_local $HELM_REPO_PATH/stable
+fi
+
+if [ -n "$asset_url" ]; then
+	for asset_url in $asset_urls; do
+	    if ! echo "$asset_url" | grep -q '.*tgz$'; then
+	        echo "Skipping $asset_url, does not look like a Helm chart archive"
+	        continue
+	    fi
+
+	    echo "Downloading $asset_url..."
+	    curl -sSfLO $asset_url
+	    # We rely on all release assets having the same baseurl
+	    download_baseurl=`dirname $asset_url`
+	done
+
+	if [ -z "$download_baseurl" ]; then
+	    echo "No Helm chart release assets found"
+	    exit 0
+	fi
+fi
+
+echo "Updating helm index"
+helm repo index stable --merge index.yaml --url https://nvidia.github.io/k8s-device-plugin/stable
```

Comment:
We're in `$HELM_REPO_PATH/stable`. Does that mean that this should be:
```
```suggestion
helm repo index . --merge index.yaml --url https://nvidia.github.io/k8s-device-plugin/stable
```